torch/distributed/checkpoint/state_dict.py

```diff
@@ -555,19 +555,34 @@ def _load_model_state_dict(
     assign = False
     if info.broadcast_from_rank0 or info.full_state_dict:
         device = None
```

Comment:
You don't need to have this `device` variable anymore. 
You can just change the line
```
device=device,
```
to
```
device=devices.pop(),
```

Reply:
Change the logic here to remove deivce and just use devices.pop() for the device input